File: frontend/src/types/heritage.ts
/**
 * M7 L5 机缘（聊天室红包）领域类型。
 */

export interface HeritageItemLine {
  item_id: string
  quantity: number
  /** 展示名（可选，列表摘要用） */
  name?: string | null
}

export interface HeritagePacket {
  id: number
  channel_type: string
  channel_ref: string
  room_id: string
  sender_character_id: number
  sender_name: string
  mode: 'random' | 'fixed' | string
  mode_label_zh: string
  share_count: number
  shares_claimed: number
  spirit_stones_total: number
  items: HeritageItemLine[]
  status: string
  note_zh: string
  expires_at: string | null
  created_at: string | null
  already_claimed: boolean
  can_claim: boolean
}

export interface HeritageClaimResult {
  message?: string
  claimed?: { spirit_stones: number; items: HeritageItemLine[] }
  packet?: HeritagePacket
  character?: Record<string, unknown>
}

File: backend/app/schemas/heritage.py
"""传承 HTTP DTO（M7 L5）。"""

from __future__ import annotations

from pydantic import BaseModel, Field


class HeritageItemLine(BaseModel):
    """传承物品行。"""

    item_id: str
    quantity: int = Field(ge=1)


class HeritageCreateRequest(BaseModel):
    """发传承。"""

    channel_ref: str
    mode: str = Field(description="random|fixed")
    share_count: int = Field(ge=1, le=100)
    spirit_stones: int = Field(default=0, ge=0)
    items: list[HeritageItemLine] = Field(default_factory=list)
    note_zh: str | None = Field(default=None, max_length=64)
